### PR TITLE
feat(common): add 234 analyticsWaveApi types

### PR DESCRIPTION
**Pull request by yijian-wang**
### What does this PR do?
Adds typings for lightning/analyticsWaveApi lds methods.

### What issues does this PR fix or reference?
@W-9717230@

cc: @smithgp

### CODE REVIEW

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (smithgp) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1149]: This is the generated type and the raml doesn't list any properties for the objects that can get added to `fields`, but it doesn't seem right. The doc links says `fields` is a `Replicated​Field​Input​Representation[]`, but `Replicated​Field​Input​Representation` isn't specified in the raml.
Maybe just switch it to `fields: Array<{ [key: string]: any }>` or similar?

**Review by smithgp — COMMENTED** [review]
cc: @mblumreich

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (yijian-wang) [reply packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1149]: Yeah, it the current implementation looks weird,  the inheritance is`ReplicatedFieldCollectionInputRepresentation -> ReplicatedFieldInputListRepresentation -> List<ReplicatedFieldInputRepresentation>. Also it seems for output representation, the generated raml cannot recognize the type on list property, i found similar issue on `WaveCollectionInputRepresentation.pins`. We do have `ReplicatedFieldInputRepresentation` in the generated raml on core side but it is never added to LDS raml.

Comment (smithgp) [reply packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1149]: After discussing this a bit, it's fine to leave this as-is for now since it doesn't appear to break anything, and there's some issues in the underlying types that we would need to address to get the right generated type showing up and we can do that later.

**Review by smithgp — APPROVED** [review]
LGTM, assuming @mblumreich is good with the verbiage.

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (randi274) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:566]: Small typo fix
```suggestion
     * Base representation for fields in Wave.
```

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (randi274) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:568]: Fixing the anchor
```suggestion
     * https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_appendix.htm#AbstractFieldRepresentation
```

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (randi274) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:687]: We should add the link to ConnectionPropertyValueRepresentation here to be consistent: https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_dataconnectors.htm#ConnectionPropertyValueRepresentation

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (randi274) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1155]:
```suggestion
     * https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_replicateddatasets_id_fields.htm#ReplicatedFieldCollectionInputRepresentation
```

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (randi274) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1167]: Nit: small typo fix
```suggestion
     * Metadata/configuration for a single field of a Replicated Dataset.
```

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (randi274) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1693]:
```suggestion
     * Wire adapter to trigger the Tableau CRM connector to run a data sync. This API is the equivalent of the “Run Now” UI feature.
```

**Comment by randi274** [comment]
Just a few small language fixes from me! @smithgp are we also waiting on review from @mblumreich?

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (mblumreich) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:613]: Change Wave to Tableau CRM

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (mblumreich) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:723]: Change Wave to Tableau CRM or just remove for "Data Connector input representation"

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (mblumreich) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1082]: Replicates data from an external source object into Tableau CRM as a dataset

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (mblumreich) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1102]: Same here

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (yijian-wang) [reply packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:613]: it seems there are many places using Wave now in this file. Do we want to change them all?

**Review by mblumreich — APPROVED** [review]
Looks good

File: packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts
Comment (mblumreich) [inline packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts:1484]: https://developer.salesforce.com/docs/component-library/documentation/en/lwc/reference_wire_adapters_get_data_connector_source_fields
(I missed this in the list I gave you)

**Review by mblumreich — APPROVED** [review]
All set, just add the url for getDataConnectorSourceFields

**Review by mblumreich — APPROVED** [review]
👍

**Comment by yijian-wang** [comment]
Hi @randi274, this PR should be good to go. Can you please review it again and then make a release on it? thanks!

**Review by randi274 — APPROVED** [review]

**Comment by randi274** [comment]
Thanks @yijian-wang - got this one in on #450. Thanks for contributing! 🎉

**Pull request closed without merging**
